Propose VaultCourier (#111)
swift-server/sswg: proposals/0034-vault-courier.md

## Maturity Justification

Sandbox. VaultCourier adheres to the minimal requirements of the SSWG Incubation Process. A stable version of the API is in development and feedback is welcomed.

## Alternatives considered

An alternative to building a Vault client with OpenAPI is to manually implement each endpoint offered by the Vault API directly in a Swift package. This is the approach taken by many clients in other languages, and also by a recent Swift package called [SwiftVault](https://swiftpackageindex.com/jmccloud827/VaultSwift).
While this method can work well, it comes with a few trade-offs:
It typically ties the client to a specific HTTP transport. For example, SwiftVault uses URLSession, which limits its compatibility with Linux, an important consideration for server-side Swift development.
It lacks built-in support for mocks, which can make development and testing more difficult.

> Manually maintaining a large and evolving API without OpenAPI can make it harder to keep the client up to date and well-documented.
> SwiftVault does appear to support many Vault features, which is commendable. I attempted to [reach out](https://github.com/jmccloud827/VaultSwift/issues/1) to the author to learn more but didn't receive a response.

Reviewer: We can tweak this a bit I think

```suggestion
SwiftVault does appear to support many Vault features. However, manually maintaining a large and evolving API without OpenAPI can make it harder to keep the client up to date and well-documented.
```
